controllers/feeds/detailFeed.js

```diff
@@ -21,6 +21,10 @@ module.exports = async (req, res) => {
     return ErrorResponse.e404(res, 'This post has been deleted');
   }
 
+  if (!feedItem) {
+    return ErrorResponse.e404(res, 'This post not found');
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The check for `feedItem` being null or undefined is a good addition to handle cases where the requested feed item does not exist. However, it would be better if this check was done before checking if the post has been deleted or expired. This way, we avoid unnecessary checks and potential errors in case `feedExpiredAt` is also undefined.

```diff
-  if (!feedItem) {
-    return ErrorResponse.e404(res, 'This post not found');
-  }
-
   if (feedExpiredAt.isBefore(moment())) {
     return ErrorResponse.e404(res, 'This post has expired and has been deleted automatically');
   }

+  if (!feedItem) {
+    return ErrorResponse.e404(res, 'This post not found');
+  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->